refactor: CarName 원시값 포장 및 테스트코드 추가

dongKos opened the pull request:
도메인 모델 패키지 추가 및 도메인 exception 추가
CarName 생성 테스트 ValueSource를 이용함

dongKos committed:
refactor: CarName 원시값 포장 및 테스트코드 추가

도메인 모델 패키지 추가 및 도메인 exception 추가
CarName 생성 테스트 ValueSource를 이용함

dongKos committed:
refactor: positon 일급 컬렉션 포장

dongKos committed:
refactor, test: Postion 포장, move interface화

Position을 불변 객체로 생성, Movable interface를 주입받아서 움직일 수 있도록 변경
Position, Car 단위 테스트 추가

chs1234 commented on src/test/java/com/dongko/domain/model/CarNameTest.java:
```diff
@@ -0,0 +1,26 @@
+class CarNameTest {
+
+    @ParameterizedTest
+    @ValueSource(strings = {"dongko", "", "null"})
```
ValueSource는 ParameterizedTest할 때 파라미터를 주입해주는 용도로 사용하는 건가요??

dongKos replied:
넵 맞습니당 테스트 메소드를 여러개 만들지 않고 테스트하는데 아주 유용하답니다.

chs1234 commented on src/test/java/com/dongko/domain/model/CarNameTest.java:
```diff
@@ -0,0 +1,26 @@
+    @ParameterizedTest
+    @ValueSource(strings = {"dongko", "", "null"})
+    @DisplayName("자동차 이름은 공백 이거나 5글자 이상은 사용 불가")
+    void test(String carName) {
```
carName에 "dongko", "", "null" 이런식으로다가?

dongKos replied:
yes. 근데 null은 바로 null타입으로 들어가진 않더라구요 ㅠ

chs1234 commented on src/main/java/com/dongko/domain/exception/RacingCarDomainException.java:
```diff
@@ -0,0 +1,9 @@
+package com.dongko.domain.exception;
+
+
+public class RacingCarDomainException extends RuntimeException {
```
RuntimeExcpetion > 도메인Exception > 하위도메인Exception
이렇게 표현하는게 맞는지 잘 모르겠지만

아 이게 "상속"이구나하는 느낌이 듭니다. 
저도 이런 방식을 좀 더 고려해야겠다는 생각이 드네요!

dongKos replied:
이렇게 하는 이유 : 
- 각 익셉션이 어떤 상황에 던져지는 건지 네이밍으로 명확히 할 수 있음
RuntimeException("잘못된 이름 입니다.");
RuntimeException("잘못된 얼굴 입니다.");

보다는
IllegalNameException();
WrongFaceException();

이런식으로 exception class 이름만 보고 알 수 있음

chs1234 commented on src/main/java/com/dongko/domain/exception/RacingCarDomainException.java:
```diff
@@ -0,0 +1,9 @@
+public class RacingCarDomainException extends RuntimeException {
+
+    public RacingCarDomainException(String message) {
+        super(message);
```
super(message) > 메시지 상위로 전파 
`
RuntimeException(String message) {
  this.message = message;
}
`

dongKos committed:
refactor, test: Cars 일급 컬렉션 포장, Participants 리팩토링, 테스트코드 추가

Cars를 일급 컬렉션으로 포장함. Participants는 cars에게 메세지를 던진다.
단위테스트를 용이하게 하기 위해 Cars에 생성자를 추가함

dongKos committed:
Merge pull request #4 from jvmoverloads/refactor/Cars-Participant

refactor, test: Cars 일급 컬렉션 포장, Participants 리팩토링, 테스트코드 추가

dongKos committed:
Merge pull request #3 from jvmoverloads/refactor/Position-and-move

refactor, test: Postion 포장, move interface화
